module/os/freebsd/zfs/zvol_os.c

```diff
@@ -1490,8 +1513,9 @@ zvol_os_create_minor(const char *name)
 out_dmu_objset_disown:
 	dmu_objset_disown(os, B_TRUE, FTAG);
 
-	if (error == 0 && volmode == ZFS_VOLMODE_GEOM) {
+	if (error == 0 && zv->zv_volmode == ZFS_VOLMODE_GEOM) {
 		g_error_provider(zv->zv_zso->zso_geom.zsg_provider, 0);
+		/* geom was locked inside zvol_alloc() function */
 		g_topology_unlock();
```

Comment:
Does not look great, but meh.

Worse here looks leak of the lock and everything else in case of `dataset_kstats_create()` error.  Not your fault, but just spotted.